shared/domain/src/main/java/com/ivy/domain/usecase/account/AccountBalanceUseCase.kt

```diff
@@ -33,11 +39,42 @@ class AccountBalanceUseCase @Inject constructor(
     suspend fun calculate(
         account: AccountId,
     ): Map<AssetCode, NonZeroDouble> {
-        TODO("Not implemented")
+
+        val transactions = transactionRepository.findAll()
+        val accountStats = accountStatsUseCase.calculate(account = account, transactions = transactions)
+
+        val balance = hashMapOf<AssetCode, NonZeroDouble>()
+
+        val income = accountStats.income
+        val transfersIn = accountStats.transfersIn
+        val expense = accountStats.expense
+        val transfersOut = accountStats.transfersOut
+
+        val amountIn = income.values.map {
+            it.key to transfersIn.values[it.key]?.value?.plus(it.value.value)
+        }.toMap()
+
+        val amountOut = expense.values.map {
+            it.key to transfersOut.values[it.key]?.value?.plus(it.value.value)
+        }.toMap()
+
+        amountIn.map {
```

Comment:
That's not mapping because it seems that you don't use the result. Replace with `.forEach`